integration-tests/src/test_loop/utils/transactions.rs

```diff
@@ -93,68 +92,69 @@ pub(crate) fn execute_money_transfers(
+/// account itself.
+///
+/// This function does not wait until the transactions is executed.
+pub fn deploy_contract(
     test_loop: &mut TestLoopV2,
     node_datas: &[TestData],
-) -> Vec<CryptoHash> {
+    rpc_id: &AccountId,
+    contract_id: &AccountId,
+) -> CryptoHash {
     let block_hash = get_shared_block_hash(node_datas, test_loop);
 
-    let mut txs = vec![];
-    for node_data in node_datas {
-        let account = node_data.account_id.clone();
-
-        let contract = near_test_contracts::rs_contract();
-        let contract_id = format!("contract.{}", account);
-        let signer = create_user_test_signer(&account).into();
-        let public_key = PublicKey::from_seed(KeyType::ED25519, &contract_id);
-        let nonce = 1;
-
-        let transaction = SignedTransaction::create_contract(
-            nonce,
-            account,
-            contract_id.parse().unwrap(),
-            contract.to_vec(),
-            10 * ONE_NEAR,
-            public_key,
-            &signer,
-            block_hash,
-        );
+    // TOOD make nonce an argument
+    let nonce = 1;
+    let signer = create_user_test_signer(&contract_id).into();
 
-        txs.push(transaction.get_hash());
+    let code = near_test_contracts::rs_contract();
+    let code = code.to_vec();
 
-        let process_tx_request =
-            ProcessTxRequest { transaction, is_forwarded: false, check_only: false };
-        let future = node_data.client_sender.clone().send_async(process_tx_request);
-        drop(future);
+    let tx = SignedTransaction::deploy_contract(nonce, contract_id, code, &signer, block_hash);
+    let tx_hash = tx.get_hash();
+    let process_tx_request =
+        ProcessTxRequest { transaction: tx, is_forwarded: false, check_only: false };
 
-        tracing::info!(target: "test", ?contract_id, "deployed contract");
-    }
-    txs
+    let rpc_node_data = get_node_data(node_datas, rpc_id);
+    let rpc_node_data_sender = &rpc_node_data.client_sender.clone();
+
+    let future = rpc_node_data_sender.send_async(process_tx_request);
+    drop(future);
+
+    tracing::debug!(target: "test", ?contract_id, ?tx_hash, "deployed contract");
+    tx_hash
 }
 
+/// Call the contract deployed at contract id from the sender id.
+///
+/// This function does not wait until the transactions is executed.
 pub fn call_contract(
     test_loop: &mut TestLoopV2,
     node_datas: &[TestData],
-    account: &AccountId,
+    sender_id: &AccountId,
+    contract_id: &AccountId,
 ) -> CryptoHash {
     let block_hash = get_shared_block_hash(node_datas, test_loop);
 
+    // TOOD make nonce an argument
     let nonce = 2;
-    let signer = create_user_test_signer(&account);
-    let contract_id = format!("contract.{}", account).parse().unwrap();
+    let signer = create_user_test_signer(sender_id);
 
     let burn_gas = 250 * TGAS;
     let attach_gas = 300 * TGAS;
 
     let deposit = 0;
+
+    // TODO make method and args arguments
```

Comment:
I assume these TODOs are intentionally not resolved yet? (this one and 2 more about nonce args)

Reply:
yeah, I don't need the full setup so I'm keeping it simple for now